### DISCUSSION TITLE
Evaluate scripts that came along with ajax-loaded content

### DISCUSSION THREAD

**Discussion by taublast** [hydrostack/hydro#71] [Ideas]
When we define our html:

```html
<div>
   ...
</div>
<script>
   ...
</script>
```

The browser evaluates the script section automatically.  

When this content is loaded in an async manner, would it be with a hydro component update or body replaced after `Location` invoked, that script will never execute.

As I recall Blazor never dealt with this 😷, maybe we could have this resolved with Hydro?

In my fork I made a [patch](https://github.com/taublast/hydro/commit/9a68fb7ecc03b5c577e6488e79bffc0b993a5a62) for this and could make a PR if acceptable.

**Comment by kjeske** [comment]
As I mentioned on one PR comment, this could be possible but would require special handling to avoid issues with CSP violations. First we can resolve the execution of custom JS triggered in actions and once that in place we can come back to this one.

**Comment by taublast** [comment]
Moved here: https://github.com/hydrostack/hydro/issues/75
